app/models/background.rb

```diff
@@ -74,10 +74,15 @@ def compute_stdev(sum, sum2, n)
   end
 
   def self.viewable(user)
+    viewable_pipeline_run_ids = PipelineRun.where(sample_id: Sample.viewable(user).pluck(:id)).pluck(:id)
     if user.admin?
       all
     else
-      where(project_id: Project.viewable(user).pluck(:id) + [nil])
+      # backgrounds which are marked as public or for which none of the pipeline_run_ids are ABSENT from the viewable_pipeline_run_ids
```

Comment:
very confusing... 
Does it meant that: "public backgrounds, for which all pipeline_run_ids are viewable"?

Reply:
No. It means:
"Background is viewable by user because
either
(a) user can view all pipeline_runs that went into the background (because he/she is part of the projects that contain them)
or
(b) background is marked as public, so any user can see it (even when they don't have access to the individual pipeline_runs that went into it).

Reply:
Ok, you should add this new explanation to the comments instead.